File: android/src/main/kotlin/com/ridehovr/hovr_app_update/VersionCompare.kt
package com.ridehovr.hovr_app_update

internal object VersionCompare {
    fun isUpdateRequired(serverVersion: String, installedVersion: String): Boolean {
        val server = serverVersion.trim()
        if (server.isEmpty()) {
            return false
        }
        return server != installedVersion.trim()
    }
}

File: ios/hovr_app_update/Sources/hovr_app_update/VersionCompare.swift
import Foundation

public enum VersionCompare {
  public static func isUpdateRequired(
    serverVersion: String,
    installedVersion: String
  ) -> Bool {
    let server = serverVersion.trimmingCharacters(in: .whitespacesAndNewlines)
    if server.isEmpty {
      return false
    }
    let installed = installedVersion.trimmingCharacters(in: .whitespacesAndNewlines)
    return server != installed
  }
}
